aptos-move/move-examples/post_mint_reveal_nft/sources/minting.move

```diff
@@ -0,0 +1,861 @@
+            let claimer_addr = signer::address_of(nft_claimer);
+            assert!(bucket_table::contains(&whitelist_mint_config.whitelisted_address, &claimer_addr), error::permission_denied(EACCOUNT_NOT_WHITELISTED));
+            let remaining_mint_allowed = bucket_table::borrow_mut(&mut whitelist_mint_config.whitelisted_address, claimer_addr);
+            assert!(*remaining_mint_allowed >= amount, error::invalid_argument(EAMOUNT_EXCEEDS_MINTS_ALLOWED));
+            while (amount > 0) {
+                mint_source_certificate_internal(nft_claimer, whitelist_mint_config.whitelist_mint_price);
+                amount = amount - 1;
+                *remaining_mint_allowed = *remaining_mint_allowed - 1;
+            };
+        } else {
+            while (amount > 0) {
+                mint_source_certificate_internal(nft_claimer, public_mint_config.public_mint_price);
+                amount = amount - 1;
+            };
+        };
+    }
+
+    // This will burn the source certificate token and put a destination token in the nft_claimer's TokenStore.
+    entry fun exchange(nft_claimer: &signer, token_id: TokenId) acquires NFTMintConfig, CollectionConfig, PublicMintConfig {
+        assert!(exists<NFTMintConfig>(@post_mint_reveal_nft) && exists<CollectionConfig>(@post_mint_reveal_nft) && exists<PublicMintConfig>(@post_mint_reveal_nft), error::permission_denied(ECONFIG_NOT_INITIALIZED));
+
+        let public_mint_config = borrow_global<PublicMintConfig>(@post_mint_reveal_nft);
+        assert!(timestamp::now_seconds() > public_mint_config.public_minting_end_time, error::permission_denied(ECANNOT_EXCHANGE_BEFORE_MINTING_ENDS));
+        assert!(token::balance_of(signer::address_of(nft_claimer), token_id) > 0, error::invalid_argument(ETOKEN_ID_NOT_FOUND));
+
+        let now = timestamp::now_microseconds();
+        let nft_mint_config = borrow_global_mut<NFTMintConfig>(@post_mint_reveal_nft);
+        let token_config = borrow_global_mut<CollectionConfig>(@post_mint_reveal_nft);
+
+        // Assert there's still some token uris in the vector.
+        assert!(vector::length(&token_config.token_uri) > 0, error::resource_exhausted(ENO_ENOUGH_TOKENS_LEFT));
+
+        let index = now % vector::length(&token_config.token_uri);
+
+        let uri = vector::swap_remove(&mut token_config.token_uri, index);
+        let keys = vector::swap_remove(&mut token_config.property_keys, index);
+        let values = vector::swap_remove(&mut token_config.property_values, index);
+        let types = vector::swap_remove(&mut token_config.property_types, index);
+
+        let token_config = borrow_global_mut<CollectionConfig>(@post_mint_reveal_nft);
+        let token_name = token_config.token_name_base;
+        string::append_utf8(&mut token_name, b": ");
+        let (owner, collection, name, property_version) = token::get_token_id_fields(&token_id);
+        let num = u64_to_string(property_version);
+        string::append(&mut token_name, num);
```

Comment:
Why?

Reply:
for the (destination) token name to have unique names and show up as `{base_token_name}: 1`, `{base_token_name}: 2` etc.

Reply:
It seems oddly manual/specific somehow

Reply:
yeah I think it seems manual, but we ideally don't want all tokens to have the same name?

Reply:
and if we aren't adding the number, it'll be hard for people to search for the token right?

Reply:
Where do the numbers for the NFTs we've seen (like Aptomingo #1201) coming from? Are they manually added like this>

Reply:
I'd think it's hardcoding but not too sure... @areshand any ideas?

Reply:
yes, the number is generated similarly. This is to reduce gas cost for storing a name for each NFT.